rfcs/0001-shabka-usage-design.md

```diff
@@ -0,0 +1,186 @@
+
+The planned use cases are very simple for now, and can be described as this:
+
+* Usage on a NixOS system, with root access.
+* Management of several hosts' configuration
+* Management of several users per-host.
+* Management of several users' configuration per-host.
+
+The expected outcome is a tool well-documented, easy to use, that can fit easily
+inside an existing workflow of managing NixOS and home-manager configurations.
+
+# Detailed design
+[design]: #detailed-design
+
+We'll discuss the three aspects previously introduced in [Summary](#summary).
+
+## `shabka` CLI
+
+The `shabka` CLI will be used by the user to do what he can currently do using
+the tools is the `scripts/` directory. This will allow him to do everything he
+could do using the `nixos-rebuild` command, plus building a host, checking what
+changed in his host's generation between two git refs, and maybe even pushing
+his hosts' configurations to Cachix.
+
+Here is what the CLI usage might look like:
+
+### `shabka rebuild`
+
+This will do what `scripts/nixos-rebuild.sh` currently does. Its usage is:
+
+```sh
+shabka rebuild [-h host] [-r release] {nixos-rebuild arguments and options}
+```
+
+By default the `host` argument is the current hostname, the `release` argument
+is the host's release (defined in `.shabka/hosts/{host}/.release`) if
+defined, or `shabka`'s default, defined in `shabka/.release`.
+
+This command takes care of setting all environment variables and options to
+ensure a smooth `nixos-rebuild` depending on what is defined in `shabka`,
+`.shabka` and `dotshabka-user`. The default locations of `.shabka` and
+`dotshabka-user` are defined in their respective section of this document.
+
+### `shabka build`
+
+This will do what `scripts/build-host.sh` currently does. Its usage is:
+
+```sh
+shabka build [-r release] [host]
+```
+
+By default the `host` argument is the current hostname, the `release` argument
+is the host's release (defined in `.hosts/hosts/{host}/.release`) if
+defined, or `shabka`'s default, defined in `shabka/.release`.
+
+> This command cannot build a host with a different kernel than the host it is
+> being ran on, e.g. a `Darwin` host cannot be built on a `Linux` host.
+
+This command takes care of setting all environment variables and options to
+ensure a smooth `nix-build` depending on what is defined in `shabka`, `.shabka`
+and `dotshabka-user`. The default locations of `.shabka` and `dotshabka-user`
+are defined in their respective section of this document.
+
+### `shabka diff`
+
+This will do what `scripts/diff-host.sh` used to do. Its usage is:
+
+```sh
+shabka diff <host> <base-rev> [target-rev]
+```
+
+By default the `target-rev` argument is HEAD.
+
+/!\ This section needs improvement given the fact that we now have 2
+repositories to deal with (`shabka` and `.shabka`, as `dotshabka-user` is
+defined in `.shabka`). This implies a lot of arguments and some designing
+around this command must be done.
```

Comment:
I'm ok to leave this command until later. We can focus on the essentials for now.

Reply:
Added a comment about this, but leaving it here for reference.